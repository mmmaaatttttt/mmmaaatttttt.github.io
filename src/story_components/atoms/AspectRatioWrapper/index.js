import PropTypes from "prop-types";
import styled from "styled-components";
import { rhythm } from "utils/typography";

const AspectRatioWrapper = styled.div`
  position: relative;
  width: 100%;
  height: 0;
  padding: ${props => props.heightOverWidth * 100}% ${rhythm(1)} 0;
  margin: 0 auto ${rhythm(1)};
`;

AspectRatioWrapper.propTypes = {
  heightOverWidth: PropTypes.number.isRequired
};

AspectRatioWrapper.defaultProps = {
  heightOverWidth: 9 / 16
};

export default AspectRatioWrapper;
