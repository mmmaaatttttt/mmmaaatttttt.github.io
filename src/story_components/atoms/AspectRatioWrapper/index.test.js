import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AspectRatioWrapper from ".";

const wrap = (props = {}) => shallow(<AspectRatioWrapper {...props} />);

it("renders successfully", () => {
  const wrapper = wrap();
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("renders children when passed in successfully", () => {
  const wrapper = shallow(
    <AspectRatioWrapper>
      <p>Hello World</p>
    </AspectRatioWrapper>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("renders props when passed in", () => {
  const wrapper = wrap({ heightOverWidth: 3 / 4 });
  expect(toJson(wrapper)).toMatchSnapshot();
});
