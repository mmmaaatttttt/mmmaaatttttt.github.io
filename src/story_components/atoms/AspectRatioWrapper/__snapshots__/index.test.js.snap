// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders children when passed in successfully 1`] = `
<div
  className="sc-bdVaJa fYjplJ"
>
  <p>
    Hello World
  </p>
</div>
`;

exports[`renders props when passed in 1`] = `
<div
  className="sc-bdVaJa fxsijT"
/>
`;

exports[`renders successfully 1`] = `
<div
  className="sc-bdVaJa fYjplJ"
/>
`;
